net/usrsock/usrsock_recvfrom.c

```diff
@@ -424,6 +424,15 @@ ssize_t usrsock_recvfrom(FAR struct socket *psock, FAR void *buf, size_t len,
                */
 
               outaddrlen = state.valuelen_nontrunc;
+
+              /* If the MSG_PEEK flag is enabled, it will only peek
+               * from the buffer, so remark the input as ready.
+               */
+
+              if (flags & MSG_PEEK)
+                {
+                  conn->flags |= USRSOCK_EVENT_RECVFROM_AVAIL;
```

Comment:
should we modify recvfrom_event to skip clear USRSOCK_EVENT_RECVFROM_AVAIL in the first place?

Reply:
> should we modify recvfrom_event to skip clear USRSOCK_EVENT_RECVFROM_AVAIL in the first place?

@xiaoxiang781216 

I'm confused that there are many 'flags' in the source code...

In the recvfrom_event(), is it possible to access the flags (i.e. MSG flags) which was passed in usrsock_recvfrom()?

Reply:
You are right, I mess up with these 'flags':(, the patch looks good now.

Reply:
@masayuki2009 Thank you for your help!

I discussed with my team members for better implementation, but there is no idea for now.